### PR TITLE
Perform dead storage removal linearly

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In crdHandler#removeDeadStorage, there is nested loop to find unused crdInfo which is torn down.

This PR restructures the loop in such a way that allCustomResourceDefinitions is traversed once to find referenced crdInfo.
storageMap is traversed one more time to tear down unused crdInfo.

**Which issue(s) this PR fixes**:
Fixes #

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go [hunk 1]
```diff
@@ -526,29 +526,20 @@ func (r *crdHandler) removeDeadStorage() {
 	r.customStorageLock.Lock()
 	defer r.customStorageLock.Unlock()
 
-	oldInfos := []*crdInfo{}
 	storageMap := r.customStorage.Load().(crdStorageMap)
-	// Copy because we cannot write to storageMap without a race
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go]: preserve the comment explaining why we make a copy to store

**Comment by liggitt** [comment]
one comment, then lgtm
/priority backlog

**Comment by tedyu** [comment]
@liggitt 
Your comment has been addressed.

**Pull request merged**